core/templates/pages/preferences-page/form-fields/preferred-languages.component.spec.ts

```diff
@@ -87,9 +80,25 @@ describe('Preferred Languages Component', () => {
       id: 'en',
       text: 'English'
     }];
+    componentInstance.filteredChoices = [{
+      id: 'en',
+      text: 'English'
+    }];
     expect(componentInstance.validInput('en')).toBeTrue();
   });
 
+
+  it('should filtered choices when input box is clicked', () => {
+    const mockChoices = [
+      { id: 'en', text: 'English ' },
+      { id: 'fr', text: 'French ' }
+    ];
+    componentInstance.choices = [...mockChoices];
+    componentInstance.onInputBoxClick();
+    expect(componentInstance.searchQuery).toEqual('');
+    expect(componentInstance.filteredChoices).toEqual(mockChoices);
+  });
+
```

Comment:
I don't see a test where your search query is non empty and filtered choices are a subset of mock choices. Similarly one where the search query doesn't match any of the available choices

Reply:
Done

Reply:
@Adarsh7825 sorry for the delay. I still don't see tests for the above 2 cases. Can you please point me to them?

Also please let the reviewers resolve conversations. It makes it easier for them to track the requested changes during a review.